lego-powered-up/src/iodevice/definition.rs

```diff
@@ -0,0 +1,336 @@
+/// Models the information available about a device from
+/// the AttachedIo, PortInformation and PortModeInformation
+/// message types.
+use std::collections::BTreeMap;
+use std::fmt;
+
+use crate::notifications::*;
+use crate::IoTypeId;
+
+type ModeId = u8;
+
+#[derive(Debug, Default, Clone)]
+pub struct Definition {
+    kind: IoTypeId,
+    port: u8,
+    capabilities: Vec<Capability>,
+    mode_count: u8,
+    modes: std::collections::BTreeMap<ModeId, PortMode>,
+    valid_combos: Vec<Vec<u8>>,
+}
+impl fmt::Display for Definition {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
```

Comment:
I don't think the explicit anonymous lifetime is needed here
```suggestion
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
```